homeassistant/components/zwave_js/sensor.py

```diff
@@ -296,6 +300,56 @@ def native_unit_of_measurement(self) -> str | None:
 class ZWaveNumericSensor(ZwaveSensorBase):
     """Representation of a Z-Wave Numeric sensor."""
 
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+        entity_description: SensorEntityDescription,
+        unit_of_measurement: str | None = None,
+    ) -> None:
+        """Initialize a ZWaveNumericSensor entity."""
+        super().__init__(
+            config_entry, client, info, entity_description, unit_of_measurement
+        )
+        self._scale = self.info.primary_value.metadata.cc_specific.get(
+            CC_SPECIFIC_SCALE
+        )
+
+    @callback
+    def _handle_scale_change_on_value_updated(self, event_data: dict) -> None:
+        """Handle scale changes for this value on value updated event."""
+        value_data = event_data["args"]
+        if (
+            value_data["commandClass"] != self.info.primary_value.command_class
+            or value_data["property"] != self.info.primary_value.property_
+            or value_data.get("endpoint") != self.info.primary_value.endpoint
+            or value_data.get("propertyKey") != self.info.primary_value.property_key
+        ):
+            return
+
+        # If the scale hasn't changed, we don't need to update the entity
+        if self._scale == (
+            new_scale := self.info.primary_value.metadata.cc_specific.get(
+                CC_SPECIFIC_SCALE
+            )
+        ):
+            return
+
+        self._scale = new_scale
+
+        self._attr_native_unit_of_measurement = (
+            NumericSensorDataTemplate()
+            .resolve_data(self.info.primary_value)
+            .unit_of_measurement
+        )
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added."""
+        await super().async_added_to_hass()
+        self.info.node.on("value updated", self._handle_scale_change_on_value_updated)
```

Comment:
Can you override `ZWaveBaseEntity.on_value_update` instead?

Reply:
good call